src/cache.h

```diff
@@ -0,0 +1,42 @@
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "list.h"
+
+/*
+    T1: LRU List
+    T2: LFU List
+    B1: LRU Ghost List
+    B2: LFU Ghost List
+*/
+typedef enum { T1, B1, T2, B2 } arc_type_t;
+
+struct hashtable;
+
+typedef struct cache {
+    struct list_head *list_table[4];
+    uint32_t list_size[4];
+    struct hashtable *map;
+    uint32_t c;
+    uint32_t p;
+#if RV32_HAS(ARCACHE_INFO)
```

Comment:
Don't add `ARCACHE_INFO`. Minimize the code as possible.

Reply:
This flag is for showing cache information, such as cache hit rate and miss rate.